src/Kokkos_ArithTraits.hpp

```diff
@@ -1292,18 +1533,99 @@ class ArithTraits< ::Kokkos::complex<float> > {
   static KOKKOS_FORCEINLINE_FUNCTION val_type conj (const val_type x) {
     return ::Kokkos::conj (x);
   }
-  // static KOKKOS_FORCEINLINE_FUNCTION val_type pow (const val_type x, const val_type y) {
-  //   return ::pow (x, y);
-  // }
+  static KOKKOS_FORCEINLINE_FUNCTION val_type pow (const val_type x, const val_type y) {
```

Comment:
I feel a bit uncomfortable with us getting into the business of transcendental function implementation.  For example, I see terms like (exp(?) - 1) in the denominator below.  What can we promise about that?  Should it always be positive?  What actually needs pow and log?  Can we skip some of these?

Reply:
I'm OK with this, if we separate the new transcendental functions into a separate commit that comes after the SIMD functionality.  I do worry about getting into this business, though I realize that the other SIMD libraries do it.

Reply:
Most math functions of ``Kokkos::complex`` naively implement mathematical definition. It uses ``mag_type`` arithmetic operations and verified wtih the results computed from ``std::complex``.  I may not care for all edge cases for now. 

Probably not many people actually need complex math functions. We can skip those hand-made functions for complex values. 

Any objection from others @mrtupek @etphipps @crtrott ?

Reply:
Okay. Let's postpone this later. Complex functions are not immediately used in any application. I commented them out.